noaa_metrics/aggregate_logs.py

```diff
@@ -16,59 +20,278 @@
 )
 
 
+def get_available_memory_gb():
+    """Get available system memory in GB."""
+    return psutil.virtual_memory().available / (1024**3)
+
+
+def get_file_size_mb(filepath: Path) -> float:
+    """Get file size in MB."""
+    try:
+        return os.path.getsize(filepath) / (1024 * 1024)
+    except:
+        return 0
+
+
+def read_json_file_safe(filepath: Path) -> pd.DataFrame:
+    """Safely read a JSON file and return DataFrame."""
+    try:
+        if os.path.getsize(filepath) > 2:
```

Comment:
Why "2"?  Prefer a named constant or a comment to explain why we're checking the getsize() here.